test/passing/break_separators.ml

```diff
@@ -0,0 +1,119 @@
+      aaaaaaaaaaaaaaaaaaa;
+      aaaaaaaaaaaaaa;
+      aaaaaaaaaaaaaaaaaa;
+      aaaaaaaaaa } =
+  { aaaaaaaaaaaa= aaaaaaaaaaaaaaaaa;
+    bbbbbbbbbbbbb= bbb bb bbbbbb;
+    cccccc= cccc ccccccccccccccccccccccc }
+
+(* this is an array *)
+let length =
+  [| 0;
+    269999999999999999999999999999999999999999999999999;
+    26;
+    (* foo *) 27 (* foo *);
+    27;
+    27 |] [@foo]
+
+(* this is a list *)
+let length =
+  ([ 0;
+    14;
+    (* foo *)
+    14;
+    17 (* foo *);
+    17;
+    2777777777777777777777777777777777;
+    27 ] [@foo])
+
+[@@@ocamlformat "break-collection-expressions=wrap"]
+
+(* this is an array *)
+let length =
+  [| 0; 1; 2; 3; 4; 5; 6; 7; 8; 8; 9; 9; 10; 10; 11; 11; 12; 12; 12; 12; 13;
+     25; 25; 25; 25; 25; 25; 25; 25; 25; 26; 26; 26; 26; 26; 26; 26; 26; 26;
+     26; 26; 26; 26; 26; 26;
+     269999999999999999999999999999999999999999999999999; 26; 26; 26; 26;
+     26; 26; 26; 26; 26; 26; 26; 26; 26; 26; 26; 26; 27; 27; 27; 27; 27; 27;
+     27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27;
+     (* foo *) 27 (* foo *); 27; 27; 27; 27; 27; 27; 27; 27; 27; 28
+  |] [@foo]
+
+(* this is a list *)
+let length =
+  ([ 0; 1; 2; 3; 4; 5; 6; 7; 8; 8; 9; 9; 10; 10; 11; 11; 12; 12; 12; 12; 13;
+     13; 13; 13; 14; 14; 14; (* foo *) 14; 15; 15; 15; 15; 16; 16; 16; 16;
+     16; 16; 16; 16; 17; 17; 17; 17 (* foo *); 17; 17; 17; 17; 18; 18; 18;
+     18; 18; 18; 18; 18; 19; 19; 19; 19; 19; 19; 19; 19; 20; 20; 20; 20; 20;
+     20; 20; 20; 20; 20; 20; 26; 26; 26; 26; 26; 27; 27; 27; 27;
+     2777777777777777777777777777777777; 27; 27; 27; 27; 27; 27; 27; 27; 27;
+     27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 27; 28
+  ] [@foo])
+
+class
+  [ 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+    'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ]
+  x =
+  [ 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+    'yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy ]
+  k
+
+type ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+       'bbbbbbbbbbbbbbbbbbbbbbbbbbbbb ) a =
+  ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+    'bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb )
+  e
+
+type ( 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+       'bbbbbbbbbbbbbbbbbbbbbbbbbbbbb ) a =
+  ('aaaaaaaaa, 'bbbbbbbbbbbb) e
+
+let ( xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+      yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy,
+      zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz,
+      (aaaaaaaaaaaa, bbbbbbbbbbbb) ) =
+  ( ( xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
+      yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy,
+      zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz ),
+    (aaaaaaaaaaaaaa, bbbbbbbbbbbb) )
+
+type t = aaaaaaaaaaaa -> bbbbbbbbbbbb -> cccccccccc
+
+type t =
+  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ->
+  bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ->
+  ccccccccccccccccccccccccc
+
+[@@@ocamlformat "ocp-indent-compat"]
+
+type t =
+  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ->
+  bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb ->
+  ccccccccccccccccccccccccc
```

Comment:
Are these change to formatting of `->` in types acceptable in the JS profile? @hhugo